File: piscine_java/war/src/main/java/fr/epita/assistants/war/Soldier.java
package fr.epita.assistants.war;

public class Soldier extends Combatant {
    int health;
    int damage_point;
    String scream;
    public Soldier() {
        this.health = 15;
        this.damage_point = 3;
        this.scream = "No pity for losers!";
    }
    public void kill() {
        this.health = 0;
    }
    @Override
    void attack(Soldier s) {
        s.health -= this.damage_point;
    }

    @Override
    void attack(Vehicle v) {
        System.out.println("I can't fight this.");
    }

    @Override
    public void scream() {
        System.out.println(this.scream);
    }
}

File: piscine_c++/war/soldier.cc
#include "soldier.hh"

#include <iostream>

Soldier::Soldier()
{
    this->health_point = 15;
    this->damage_point = 3;
    this->crie = "No pity for losers!\n";
}

void Soldier::attack(Soldier& s)
{
    s.health_point -= this->damage_point;
}

void Soldier::print_state() const
{
    std::cout << "I have " << this->health_point << " health points.\n";
}

void Soldier::scream() const
{
    std::cout << this->crie;
}
